src/components/HomePage/AddEditBooksDialog.jsx

```diff
@@ -0,0 +1,200 @@
+import { Button, Col, Form, Modal, Row } from "react-bootstrap"
+import styles from "../styles/homePage/addEditBooksModal.module.css"
+import { useState } from "react";
+import ToastEventKeys from "../UiCommon/ToastEventKeys";
+import BookManager from "../../common/BooksManager";
+
+const AddEditBooksDialog = (props) => {
+
+    const isValidUrl = urlString => {
+        var urlPattern = new RegExp('^(https?:\\/\\/)?' +
+            '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' +
+            '((\\d{1,3}\\.){3}\\d{1,3}))' +
+            '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' +
+            '(\\?[;&a-z\\d%_.~+=-]*)?' +
+            '(\\#[-a-z\\d_]*)?$', 'i');
```

Comment:
WTF wierzę na słowo, że ten pattern jest okej bo sam go nie odczytam XD

Reply:
ja go z neta skopiowałem i działa także to najważniejsze xD